blog.md

```diff
@@ -0,0 +1,112 @@
+```py
+env = gymansium.make('Swimmer-v5', xml_file="MY_MUJOCO_MODEL.xml", ...)
+```
+
+But we will need to change some arguments to specify the behavior we want, the most commonly changed arguments are:
+- `xml_file`: Path to the MuJoCo robot.
+- `frame_skip`: to set the duration of a time step (`dt`) (recommended range of `dt` is $\[0.01, 0.1\]$). 
+- `ctrl_cost_weight`: set it according to the needs of the robot, we can set it to `0` at first for prototyping and increase it as needed.
+- `healthy_z_range`: set it according to the height of the robot.
+For more information on all the arguments, see the documentation page of `Humanoid`, `Ant` & respectively.
+
+## Using `Ant` framework to create an RL environment for a Quadruped robot
+
+### Example [anybotics_anymal_b](https://github.com/deepmind/mujoco_menagerie/blob/main/anybotics_anymal_b/README.md)
+```py
+env = gymnasium.make('Ant-v5', xml_file='./mujoco_menagerie/anybotics_anymal_b/scene.xml', ctrl_cost_weight=0.001, healthy_z_range=(0.48, 0.68), render_mode='human')
+```
+Here all we have to do is to change the `xml_file` argument, and set the `healthy_z_range`, because the robot has a different height than the default `Ant` robot, in general we will have to change the `healthy_z_range` to fit the robot.
+
+<iframe id="odysee-iframe" width="560" height="315" src="https://odysee.com/$/embed/@Kallinteris-Andreas:7/ANYmal_B_trained_using_SAC_on_gymnasium_mujoco-v5_framework:1?r=6fn5jA9uZQUZXGKVpwtqjz1eyJcS3hj3" allowfullscreen></iframe>
+ 
+ ### Example [Unitree Go1](https://github.com/deepmind/mujoco_menagerie/blob/main/unitree_go1/README.md)
+```py
+env = gym.make('Ant-v5', xml_file='./mujoco_menagerie/unitree_go1/scene.xml', healthy_z_range=(0.195, 0.75), ctrl_cost_weight=0.05)
+```
+<iframe id="odysee-iframe" width="560" height="315" src="https://odysee.com/$/embed/@Kallinteris-Andreas:7/Unitree_Go1_trained_using_SAC_on_gymnasium_mujoco-v5_framework:5?r=6fn5jA9uZQUZXGKVpwtqjz1eyJcS3hj3" allowfullscreen></iframe>
+
+## Using `Humanoid` framework to create an RL environment for a Bipeds robot
+
+
+### Example [Robotis OP3](https://github.com/deepmind/mujoco_menagerie/blob/main/robotis_op3/README.md)
+```py
+env = gym.make('Humanoid-v5', xml_file='~/mujoco_menagerie/robotis_op3/scene.xml', healthy_z_range=(0.275, 0.5), include_cinert_in_observation=False, include_cvel_in_observation=False, include_qfrc_actuator_in_observation=False, include_cfrc_ext_in_observation=False, ctrl_cost_weight=0, contact_cost_weight=0)
+```
+
+<iframe id="odysee-iframe" width="560" height="315" src="https://odysee.com/$/embed/@Kallinteris-Andreas:7/Robotis_OP3_trained_using_PPO_on_gymnasium_mujoco-v5_framework:d?r=6fn5jA9uZQUZXGKVpwtqjz1eyJcS3hj3" allowfullscreen></iframe>
+
+For more detailed tutorial check the tutorial for [loading quadruped models](https://gymnasium.farama.org/main/tutorials/gymnasium_basics/load_quadruped_model/).
+
+
+
+# Full Changelog
```

Comment:
Do you have any training-evaluation results that highlight the extended modularity of the new envs? If you're interested, I could integrate some examples into [tianshou](https://github.com/thu-ml/tianshou)

Side note: personally, I feel that the gym.make mechanism is an unfortunate choice for creating environments, especially those permitting modularity. I prefer to instantiate wrappers and write code creating envs from configuration myself, which gives type-safe and debuggable code (finding out where and how an env and it's spec are registered and how wrappers are instantiated from the spec felt like a nightmare to me). So, given this extended modularity of Mujoco envs, I would appreciate examples of how to use it without relying on `gym.make`.

If you are interested, I can send you some info on how I currently design and manage modular envs. For practical RL applications, this is a central topic

Reply:
Hey @MischaPanch, first thanks for taking time to review it

There are 2 different problems with regard to custom robot models
1) creating the environment, deciding on what environment parameter to use, (which will likely be an iterative process)
an example I have done can be found in the tutorial.
2) evaluating the locomotion performance of the created environment (which will likely require hyper tuning, or even new algorithmic techniques)


side note: the lack of type hinting in `gymnasium.make()` does indeed suck

other note: yes, I am interested sent me anything you want, but this is goes off-topic to the review of the blog post

Reply:
Cool, I'll contact you on discord.

Concerning the blog post: I just thought it'd be interesting to include how training curves change depending on the changes in the environment, thereby assessing the changed "difficulty" of the env, and further justifying the need for the new features.

I agree that a meaningful statement of this type requires proper evaluation and HPO, which is precisely what I'm incorporating now into tianshou algorithm examples. Of course, it could also be done with cleanRL or any other framework. If you feel it's off-topic for the blog, feel free to resolve the thread and we'll continue on discord :)

Reply:
The difficulty of environments is roughly the same (you can check the GitHub issues),

environments with realistic models, are not directly comparable with the pre-included simple models